assets/js/components/modules-list.js

```diff
@@ -73,11 +72,17 @@ class ModulesList extends Component {
 	}
 
 	render() {
-		const modules = window.googlesitekit.modules || [];
+		// Filter out internal modules.
+		const modules = filter( window.googlesitekit.modules || [], ( module ) => ! module.internal );
+
+		// Map of slug => name for every module that is active and completely set up.
+		const completedModuleNames = {};
 
 		// Sort Modules by sort order.
-		let sortedModules = sortBy( modules, ( module, key ) => {
-			module.key = key;
+		let sortedModules = sortBy( modules, ( module ) => {
+			if ( module.active && module.setupComplete ) {
+				completedModuleNames[ module.slug ] = module.name;
+			}
```

Comment:
`completedModuleNames` shouldn't be mutated within a callback for sorting logic.

This operation should probably be done as part if the initialization of `completedModuleNames`,
e.g
```js
const completedModuleNames = modules
	.filter( ( module ) => module.active && module.setupComplete )
	.reduce( ( completed, module ) => {
		completed[ module.slug ] = module.name;
		return completed;
	}, {} );
```

Then `sortedModules` could be simplified as well 

```js
const sortedModules = _( modules )
	.filter( ( module ) => 0 === module.required.length )
	.sortBy( 'sort' )
	.value();
```

Reply:
Makes sense, I'll update that. We can probably use native array functions `filter()`, `sort()` though right?

Reply:
Yeah, the first block is all native functions, the second could be rewritten to use native as well, we'd just need to implement the sorting.